source/Concepts/Basic/About-Launch.rst

```diff
@@ -7,30 +7,48 @@ Launch
+``launch`` works with _actions_, abstract representations of computational procedures with side effects on its execution environment. Logging to standard output, executing a program, shutting down ``launch`` itself, are examples of actions. An action may also encompass another action or a collection thereof to describe more complex functionality. A collection of actions makes up the _description_ of a procedure that ``launch`` can take and execute. These descriptions are typically read from source files, so called _launch files_, which may be written in Python, or using specific XML or YAML syntax.
 
-Actions alone allow for static descriptions that act upon and react to the environment in predefined ways. For a description to adapt to external input, _conditions_ and _substitutions_ exist. The former are boolean predicates, used for conditional actions, while the latter resemble text macros, useful for dynamic action configuration.
+While the extent of the execution environment of an action depends on its nature e.g. logging to standard output is circumscribed to the ``launch`` process whereas executing a program reaches out to the host operating system, ``launch`` maintains an execution _context_ in which these actions take place and through which these can interact between them and with the user.
+The ``launch`` context holds _configuration variables_ (i.e. state) and propagates _events_, both of which are available to actions and to ``launch`` itself.
+Events are signals that originate internally in actions or externally on user input. It is through events, for example, that ``launch`` description execution (i.e. inclusion) is triggered.
+Configuration variables populate the ``launch`` context. It is through configuration variables, for example, that ``launch`` descriptions can take arguments before execution.
 
-In a way, ``launch`` descriptions are not that different from ASTs in procedural programming languages. The analogy brings about an important distinction that is easy to miss when writing launch files using the native Python APIs: neither actions nor conditions nor substitutions are computations but descriptions of computations to be carried out on launch. That is how actions can be implemented as sequences of other actions -- launch file inclusion being the paradigmatic example of this.
+In connection with its context, ``launch`` defines _conditions_, _substitutions_, and _event handlers_ to broaden its scope of application.
+The first two are different forms of expressions evaluated in ``launch`` context (though they may also tap into the larger execution environment): conditions are boolean predicates, used to define conditional actions, while substitutions are string interpolation expressions that enable dynamic ``launch`` descriptions.
```

Comment:
Simple examples like you do with action would be helpful for each of these as well I think.

Reply:
Done in https://github.com/mhidalgo-bdai/ros2_documentation/pull/1/commits/1ecbbac0e4a1339cc85886fb9a60f50c9ede4214.